sufia-models/app/models/concerns/sufia/generic_file/csv.rb

```diff
@@ -0,0 +1,23 @@
+module Sufia
+  module GenericFile
+    module CSV
+      def to_csv
+        ::CSV.generate do |csv|
+          csv << csv_terms.map do |term|
+            values = send(term)
+            values = [values].flatten # make sure we have an array
+            values.join(multi_value_separator)
```

Comment:
@jcoyne has worked on a CSV importer that uses multiple columns with the same name to handle repeatability which is a little safer (because metadata values can contain the pipe character or any other weird character(s)). Would this be acceptable for the use case you're working on here or would that, e.g., break the way Excel imports the CSV?